hikari/impl/rest.py

```diff
@@ -351,6 +348,61 @@ def acquire(
         return rest_client
 
 
+@attr.define()
+class _LiveAttributes:
+    """Fields which are only present within `RESTClientImpl` while it's "live".
+
+    !!! note
+        This must be started within an active asyncio event loop.
+    """
+
+    buckets: buckets_.RESTBucketManager
+    client_session: aiohttp.ClientSession
+    closed_event: asyncio.Event
+    # We've been told in DAPI that this is per token.
+    global_rate_limit: rate_limits.ManualRateLimiter
+    tcp_connector: aiohttp.TCPConnector
+
+    @classmethod
+    def build(
+        cls, max_rate_limit: float, http_settings: config.HTTPSettings, proxy_settings: config.ProxySettings
+    ) -> _LiveAttributes:
+        """Build a live attributes object.
+
+        !!! warning
+            This can only be called when the current thread has an active
+            asyncio loop.
+        """
+        # This asserts that this is called within an active event loop.
+        asyncio.get_running_loop()
+        tcp_connector = net.create_tcp_connector(http_settings)
```

Comment:
does TCP connector have to be handled this way or can it be persisted between starts and initiated in `RESTClientImpl.__init__`?

Reply:
Tbf don't really care